lib/querystring.js

```diff
@@ -209,18 +209,20 @@ QueryString.parse = QueryString.decode = function(qs, sep, eq, options) {
   }
 
   var regexp = /\+/g;
-  qs = qs.split(sep);
 
   var maxKeys = 1000;
```

Comment:
It's in [the docs](https://nodejs.org/api/querystring.html#querystring_querystring_parse_str_sep_eq_options), changing it would mean a major version upgrade. I don't know the reasoning behind it, I guess 1000 is a safe but large enough number of key value pairs.